Sanity check fails when the error message depends on the presence of the inductive in AdmitTactic

Consider
```coq
Module Foo.
Module Export AdmitTactic.
Module Import LocalFalse.
Inductive False := .
End LocalFalse.
End AdmitTactic.
Goal AdmitTactic.LocalFalse.False.
  exact I.
```
This fails with a log ending with
```
Now, I will attempt to strip repeated newlines and trailing spaces from this file...

No strippable newlines or spaces.
getting foo_01.v
coq_makefile COQC = coqc -o Makefile2bPitL.coq -R . Top -R . Top -arg -top -arg foo foo.v foo_01.v inconsistency.v inconsistency_01.v tmpVNTbbE.v
Warning: . and . overlap (used in -R or -Q)

make -k -f Makefile2bPitL.coq foo_01.glob
COQDEP VFILES
coqc  -q -top foo  -R . Top -R . Top foo_01.v
File "./foo_01.v", line 11, characters 8-9:
Error:
The term "I" has type "True" while it is expected to have type
 "LocalFalse.False".

Makefile2bPitL.coq:663: recipe for target 'foo_01.glob' failed
make: *** [foo_01.glob] Error 1
getting foo_01.v
WARNING: Assuming that foo_01.glob is not a valid reflection of foo_01.v because foo_01.v is newer
coq_makefile COQC = coqc -o MakefileXJ6xOq.coq -R . Top -R . Top -arg -top -arg foo foo.v foo_01.v inconsistency.v inconsistency_01.v tmpVNTbbE.v
Warning: . and . overlap (used in -R or -Q)

make -k -f MakefileXJ6xOq.coq foo_01.glob
COQDEP VFILES
coqc  -q -top foo  -R . Top -R . Top foo_01.v
File "./foo_01.v", line 15, characters 0-26:
Error: The label AdmitTactic is already declared.

MakefileXJ6xOq.coq:663: recipe for target 'foo_01.glob' failed
make: *** [foo_01.glob] Error 1

Non-fatal error: Failed to validate all coq runs and preserve the error.
The new error was:
File "/tmp/tmppzPDoU.v", line 15, characters 0-26:
Error: The label AdmitTactic is already declared.


File not saved.
Fatal error: Sanity check failed.
```

This is a perverse example (that should work anyway), but this came up in the wild for 
```coq
Require Coq.Init.Notations.
Module Export Mtac2_DOT_lib_DOT_Datatypes.
Module Export Mtac2.
Module Export lib.
Module Export Datatypes.

Set Implicit Arguments.

Set Universe Polymorphism.

Inductive munit : Set :=
    mtt : munit.

Cumulative Inductive moption (A:Type) : Type :=
  | mSome : A -> moption A
  | mNone : moption A.
Arguments mNone {A}.

Definition moption_map (A B:Type) (f:A->B) (o : moption A) : moption B :=
  match o with
    | mSome a => @mSome B (f a)
    | mNone => @mNone B
  end.

Inductive mprod (A B:Type) : Type :=
  mpair : A -> B -> mprod A B.

Module Export ProdNotations.
Infix "*m" := (mprod) (at level 40) : type_scope.
Notation "(m: x , y , .. , z )" := (mpair .. (mpair x y) .. z) : core_scope.
End ProdNotations.

Section projections.
  Context {A : Type} {B : Type}.

  Definition mfst (p:A *m B) := match p with
				| (m: x, y) => x
                              end.
  Definition msnd (p:A *m B) := match p with
				| (m: x, y) => y
                              end.
End projections.

Cumulative Inductive mlist (A : Type) : Type :=
 | mnil : mlist A
 | mcons : A -> mlist A -> mlist A.

Arguments mnil {A}.
Infix ":m:" := mcons (at level 60, right associativity) : mlist_scope.

Local Open Scope mlist_scope.

Definition mlength (A : Type) : mlist A -> nat :=
  fix length l :=
  match l with
   | mnil => O
   | _ :m: l' => S (length l')
  end.

Definition mapp (A : Type) : mlist A -> mlist A -> mlist A :=
  fix mapp l m :=
  match l with
   | mnil => m
   | a :m: l1 => a :m: mapp l1 m
  end.

Infix "+m+" := mapp (right associativity, at level 60) : mlist_scope.

End Datatypes.

End lib.

End Mtac2.

End Mtac2_DOT_lib_DOT_Datatypes.

Module Import LocalFalse.
Inductive False := .
End LocalFalse.
Module Export Mtac2_DOT_lib_DOT_List.
Module Export Mtac2.
Module Export lib.
Module Export List.
Set Implicit Arguments.

Set Universe Polymorphism.

Open Scope mlist_scope.

Module Export ListNotations.
Notation "[m: ]" := mnil (format "[m: ]") : mlist_scope.
Notation "[m: x ]" := (mcons x mnil) : mlist_scope.
End ListNotations.

Definition mhd {A} (default:A) (l:mlist A) :=
  match l with
  | [m:] => default
  | x :m: _ => x
  end.

Definition mhd_error {A} (l:mlist A) : moption A :=
  match l with
  | [m:] => mNone
  | x :m: _ => mSome x
  end.

Definition mtl {A} (l:mlist A) :=
  match l with
  | [m:] => mnil
  | a :m: m => m
  end.

Fixpoint mnth {A} (n:nat) (l:mlist A) (default:A) {struct l} : A :=
  match n, l with
  | O, x :m: l' => x
  | O, other => default
  | S m, [m:] => default
  | S m, x :m: t => mnth m t default
  end.

Fixpoint mnth_ok {A} (n:nat) (l:mlist A) (default:A) {struct l} : bool :=
  match n, l with
  | O, x :m: l' => true
  | O, other => false
  | S m, [m:] => false
  | S m, x :m: t => mnth_ok m t default
  end.

Fixpoint mnth_error {A} (l:mlist A) (n:nat) {struct n} : moption A :=
  match n, l with
  | O, x :m: _ => mSome x
  | S n, _ :m: l => mnth_error l n
  | _, _ => mNone
  end.

Definition mnth_default {A}(default:A) (l:mlist A) (n:nat) : A :=
  match mnth_error l n with
  | mSome x => x
  | mNone => default
  end.

Fixpoint mlast {A} (l:mlist A) (d:A) : A :=
  match l with
  | [m:] => d
  | [m:a] => a
  | a :m: l => mlast l d
  end.

Fixpoint mremovemlast {A} (l:mlist A) : mlist A :=
  match l with
  | [m:] =>  [m:]
  | [m:a] => [m:]
  | a :m: l => a :m: mremovemlast l
  end.

Fixpoint mrev {A} (l:mlist A) : mlist A :=
  match l with
  | [m:] => [m:]
  | x :m: l' => mrev l' +m+ [m:x]
  end.

Fixpoint mrev_append {A} (l l': mlist A) : mlist A :=
  match l with
  | [m:] => l'
  | a:m:l => mrev_append l (a:m:l')
  end.

Definition mrev' {A} l : mlist A := mrev_append l [m:].

Fixpoint mconcat {A} (l : mlist (mlist A)) : mlist A :=
  match l with
  | mnil => mnil
  | mcons x l => x +m+ mconcat l
  end.

Definition mmap {A B} (f : A -> B) :=
  fix mmap (l:mlist A) : mlist B :=
  match l with
    | [m:] => [m:]
    | a :m: t => (f a) :m: (mmap t)
  end.

Definition mflat_mmap {A B} (f:A -> mlist B) :=
  fix mflat_mmap (l:mlist A) : mlist B :=
  match l with
    | mnil => mnil
    | mcons x t => (f x)+m+(mflat_mmap t)
  end.

Section Fold_Left_Recursor.
  Variables (A : Type) (B : Type).
  Variable f : A -> B -> A.

  Fixpoint mfold_left (l:mlist B) (a0:A) : A :=
    match l with
    | mnil => a0
    | mcons b t => mfold_left t (f a0 b)
    end.
End Fold_Left_Recursor.

Section Fold_Right_Recursor.
  Variables (A : Type) (B : Type).
  Variable f : B -> A -> A.
  Variable a0 : A.

  Fixpoint mfold_right (l:mlist B) : A :=
    match l with
    | mnil => a0
    | mcons b t => f b (mfold_right t)
    end.
End Fold_Right_Recursor.

Section Bool.
  Variable A : Type.
  Variable f : A -> bool.

  Fixpoint mexistsb (l:mlist A) : bool :=
    match l with
    | mnil => false
    | a:m:l => f a || mexistsb l
    end.

  Fixpoint mforallb (l:mlist A) : bool :=
    match l with
    | mnil => true
    | a:m:l => f a && mforallb l
    end.

  Fixpoint mfilter (l:mlist A) : mlist A :=
    match l with
    | mnil => mnil
    | x :m: l => if f x then x:m:(mfilter l) else mfilter l
    end.

  Fixpoint mfind (l:mlist A) : moption A :=
    match l with
    | mnil => mNone
    | x :m: mtl => if f x then mSome x else mfind mtl
    end.

  Fixpoint mpartition (l:mlist A) : mlist A * mlist A :=
    match l with
    | mnil => (mnil, mnil)
    | x :m: mtl => let (g,d) := mpartition mtl in
      if f x then (x:m:g,d) else (g,x:m:d)
      end.
End Bool.

Fixpoint msplit {A B} (l:mlist (A*B)) : mlist A * mlist B :=
  match l with
  | [m:] => ([m:], [m:])
  | (x,y) :m: mtl => let (left,right) := msplit mtl in (x:m:left, y:m:right)
  end.

Fixpoint mcombine {A B} (l : mlist A) (l' : mlist B) : mlist (A*B) :=
  match l,l' with
  | x:m:mtl, y:m:mtl' => (x,y):m:(mcombine mtl mtl')
  | _, _ => mnil
  end.

Fixpoint mlist_prod {A B} (l:mlist A) (l':mlist B) : mlist (A * B) :=
  match l with
  | mnil => mnil
  | mcons x t => (mmap (fun y:B => (x, y)) l')+m+(mlist_prod t l')
  end.

Fixpoint mfirstn {A} (n:nat)(l:mlist A) : mlist A :=
  match n with
  | 0 => mnil
  | S n => match l with
	  | mnil => mnil
	  | a:m:l => a:m:(mfirstn n l)
    end
  end.

Fixpoint mskipn {A} (n:nat)(l:mlist A) : mlist A :=
  match n with
  | 0 => l
  | S n => match l with
	  | mnil => mnil
	  | a:m:l => mskipn n l
    end
  end.

Fixpoint mseq (start len:nat) : mlist nat :=
  match len with
  | 0 => mnil
  | S len => start :m: mseq (S start) len
  end.

Fixpoint mrepeat {A} (x : A) (n: nat ) :=
  match n with
  | O => [m:]
  | S k => x:m:(mrepeat x k)
  end.

End List.

End lib.

End Mtac2.

End Mtac2_DOT_lib_DOT_List.

Inductive False := .
Module Export Mtac2.
Module Export Sorts.
Set Universe Polymorphism.
Module S.

Monomorphic Inductive Sort : Type := Prop_sort | Type_sort.

Notation "'Typeₛ'" := Type_sort.
Notation "'Propₛ'" := Prop_sort.

Definition stype_of (s : Sort) : Type :=
  match s with Typeₛ => Type | Propₛ => Prop end.

Definition selem_of@{i j+} {s : Sort} : stype_of@{i j} s -> Type@{j} :=
  match s return stype_of s -> Type@{j} with
  | Typeₛ => fun x => x
  | Propₛ => fun x => x
  end.

Fail Local Example CannotMakeAnElementOfaSort s (P : stype_of s) (x : P) := x.
Local Example WeCanWithElemOf s (P : stype_of s) (x : selem_of P) := x.

Definition selem_lift {s : Sort} : @selem_of Typeₛ (stype_of s) -> Type :=
  match s as s' return @selem_of Typeₛ (stype_of s') -> Type with
  | Typeₛ => fun x => x
  | Propₛ => fun y => y
  end.

Definition ForAll
            {sort : Sort} {A : Type} :
  (A -> stype_of sort) -> stype_of sort :=
  match
    sort as sort'
    return ((A -> stype_of sort') -> stype_of sort')
  with
  | Propₛ => fun F => forall a : A, F a
  | Typeₛ => fun F => forall a : A, F a
  end.

Definition Impl {sort : Sort} A (B : stype_of sort) : stype_of sort :=
  ForAll (sort := sort) (fun _ : A => B).

Definition Fun {sort} {A : Type} :
  forall {F : A -> stype_of sort}, (forall a, selem_of (F a)) -> selem_of (ForAll F) :=
  match sort as sort' return
        forall {F : A -> stype_of sort'}, (forall a, selem_of (F a)) -> selem_of (ForAll F)
  with
  | Propₛ => fun _ f => f
  | Typeₛ => fun _ f => f
  end.

Definition App {sort} {A : Type} : forall {F : A -> _},  selem_of (ForAll (sort := sort) F) -> forall a, selem_of (F a) :=
  match sort as sort' return forall F, selem_of (ForAll (sort := sort') F) -> forall a, selem_of (F a) with
  | Propₛ => fun F f a => f a
  | Typeₛ => fun F f a => f a
  end.

Definition Impl_lift {sort} {A : Type} : forall {B : stype_of sort}, (A -> selem_of B) -> selem_of (Impl A B) :=
  match sort with
  | Propₛ => fun B f => f
  | Typeₛ => fun B f => f
  end.

Definition lift_Impl {sort} {A : Type} : forall {B : stype_of sort}, selem_of (Impl A B) -> (A -> selem_of B) :=
  match sort with
  | Propₛ => fun B f => f
  | Typeₛ => fun B f => f
  end.

End S.

Import S.
Notation "'Typeₛ'" := Type_sort.

Coercion stype_of : Sort >-> Sortclass.
Coercion selem_of : stype_of >-> Sortclass.

End Sorts.

Inductive False := .

Set Implicit Arguments.

Set Universe Polymorphism.

Inductive msig (A:Type) (P:A -> Prop) : Type :=
    mexist : forall x:A, P x -> msig P.

Inductive msig2 (A:Type) (P Q:A -> Prop) : Type :=
    mexist2 : forall x:A, P x -> Q x -> msig2 P Q.

Inductive msigT (A:Type) (P:A -> Type) : Type :=
    mexistT : forall x:A, P x -> msigT P.

Inductive msigT2 (A:Type) (P Q:A -> Type) : Type :=
    mexistT2 : forall x:A, P x -> Q x -> msigT2 P Q.

Section Subset_mprojections.

  Variable A : Type.
  Variable P : A -> Prop.

  Definition mproj1_msig (e:msig P) := match e with
                                    | mexist _ a b => a
                                    end.

End Subset_mprojections.

Definition msig_of_msig2 (A : Type) (P Q : A -> Prop) (X : msig2 P Q) : msig P
  := mexist P
           (let (a, _, _) := X in a)
           (let (x, p, _) as s return (P (let (a, _, _) := s in a)) := X in p).

Section Subset_mprojections2.

  Variable A : Type.
  Variables P Q : A -> Prop.

  Definition mproj3_msig (e : msig2 P Q) :=
    let (a, b, c) return Q (mproj1_msig (msig_of_msig2 e)) := e in c.

End Subset_mprojections2.

Section Mprojections.

  Variable A : Type.
  Variable P : A -> Type.

  Definition mprojT1 (x:msigT P) : A := match x with
                                      | mexistT _ a _ => a
                                      end.

End Mprojections.

Definition msigT_of_msigT2 (A : Type) (P Q : A -> Type) (X : msigT2 P Q) : msigT P
  := mexistT P
            (let (a, _, _) := X in a)
            (let (x, p, _) as s return (P (let (a, _, _) := s in a)) := X in p).

Module Export MTele.
Import Sorts.S.
Unset Universe Minimization ToSet.
Set Polymorphic Inductive Cumulativity.

Inductive MTele : Type :=
| mBase : MTele
| mTele {X : Type} (F : X -> MTele) : MTele
.

Fixpoint MTele_Const {s : Sort} (T : s) (n : MTele) : s :=
match n with
| mBase => T
| mTele F => ForAll (fun x => MTele_Const T (F x))
end.
Definition MTele_ConstT (T : Type) (n : MTele) : Type := @MTele_Const Typeₛ T n.

Definition MTele_Sort (s : Sort) (n : MTele) : Type := MTele_ConstT (stype_of s) n.

Fixpoint MTele_sort {s : Sort} {n : MTele} :
  forall S : MTele_Sort s n, Type :=
  match n return MTele_Sort s n -> _ with
  | mBase => fun S => selem_of S
  | mTele F => fun S => forall x, MTele_sort (S x)
  end.

Fixpoint MTele_val {s} {n : MTele} : MTele_Sort s n -> s :=
  match n as n return MTele_Sort s n -> s with
  | mBase => fun f => f
  | mTele F => fun f => ForAll (fun x => MTele_val (f x))
  end.

Fixpoint ArgsOf (m : MTele) : Type :=
  match m with
  | mBase => unit
  | mTele f => msigT (fun x => ArgsOf (f x))
  end.

Fixpoint apply_const {s : Sort} {m : MTele} {T : s} :
  MTele_Const T m -> ArgsOf m -> T :=
  match m with
  | mBase => fun t _ => t
  | mTele f => fun t '(mexistT _ x U) => apply_const (App t x) U
  end.
Definition apply_constT {m T} := @apply_const Typeₛ m T.

Definition apply_sort {s : Sort} {m : MTele} :
  MTele_Sort s m -> ArgsOf m -> stype_of s :=
  @apply_const Typeₛ m (stype_of s).

Fixpoint apply_val {s : Sort} {m : MTele} :
  forall {T : MTele_Sort s m} (v : MTele_val T) (a : ArgsOf m), apply_sort T a :=
  match m with
  | mBase => fun _ v _ => v
  | mTele f => fun _ v '(mexistT _ x U) => apply_val (App v x) U
  end.

Fixpoint curry_const {s : Sort} {m : MTele} {T : s} : (ArgsOf m -> T) -> MTele_Const T m :=
  match m with
  | mBase => fun f => f tt
  | mTele F => fun f => Fun (fun x => curry_const (fun a => f (mexistT (fun x => ArgsOf _) x a)))
  end.

Definition curry_sort (s : Sort) {m : MTele} : _ -> MTele_Sort s m :=
  @curry_const Typeₛ m (stype_of s).

Fixpoint apply_curry_sort {s} {m} :
  forall {f : ArgsOf m -> _} {a : ArgsOf m}, apply_sort (curry_sort s f) a -> selem_of (f a) :=
  match m as m return
        forall {f : ArgsOf m -> _} {a : ArgsOf m}, apply_sort (curry_sort s f) a -> selem_of (f a)
  with
  | mBase => fun f 'tt t => t
  | mTele F => fun f '(mexistT _ x a) t => @apply_curry_sort _ (F x) (fun args => f (mexistT _ x args)) _ t
  end.

End MTele.
Require Import Coq.Strings.String.

Record ind_sig {params : MTele} : Type :=
  {
    ind_sig_sort : S.Sort;
    ind_sig_arity : MTele_ConstT MTele params;
  }.
Arguments ind_sig _ : clear implicits.

Record ind_def {params: MTele} : Type :=
  { ind_def_name : string; ind_def_sig : ind_sig params }.
Arguments ind_def _ : clear implicits.

Definition ind_arg {params} (i : ind_def params) : Type :=
  let sort := ind_sig_sort (ind_def_sig i) in
  let arity := ind_sig_arity (ind_def_sig i) in
    MTele_val (curry_sort Typeₛ
                 (fun a' => MTele_Sort sort (apply_constT arity a'))
              ).

Record constr_def_wop {params} {ind : MTele_ConstT MTele params} :=
  {
    constr_def_wop_name : string;
    constr_def_wop_tele : MTele_ConstT MTele params;
    constr_def_wop_indices : MTele_val (curry_sort Typeₛ (fun args => MTele_ConstT (ArgsOf (apply_constT ind args)) (apply_constT constr_def_wop_tele args)));
  }.
Arguments constr_def_wop {_} _.

Definition constr_def_value_wop
           {params}
           {i : ind_def params}
           (ia: ind_arg i)
           (c : constr_def_wop (ind_sig_arity (ind_def_sig i))) : Type :=
  MTele_val (
      curry_sort
        Typeₛ
        (fun p_args =>
           let c_params := apply_constT (constr_def_wop_tele c) p_args in
           let c_inds := apply_val (s:=Typeₛ) (constr_def_wop_indices c) p_args in
           let c_inds := apply_curry_sort c_inds in
           let ia := apply_val (s:=Typeₛ) ia p_args in
           let ia := apply_curry_sort ia in
           MTele_sort (
               curry_sort
                 _
                 (fun c_args =>
                    let args := apply_constT c_inds c_args in
                    let ind := apply_sort ia args in
                    ind
                 )
             )
        )
    ).

Definition constrs_def_wop {params} (ind : MTele_ConstT MTele params) :=
  mlist (constr_def_wop ind).

Definition return_predicate_type
           {params}
           {i : ind_def params}
           (p_args : ArgsOf params)
           (ia : ind_arg i)
           (s : S.Sort)
  : Type :=
    MTele_val
      (
        curry_sort Typeₛ (fun i_args => apply_sort (apply_curry_sort (apply_val (s:=Typeₛ) ia p_args)) i_args -> s)
      ).

Set Printing All.
Set Printing Universes.
Record Match :=
  {
    match_sort: S.Sort;
    match_param_tele: MTele;
    match_ind_def: ind_def match_param_tele;
    match_ind : ind_arg match_ind_def;
    match_constrs_sig: constrs_def_wop (ind_sig_arity (ind_def_sig match_ind_def));
    match_constrs: (mfold_right (fun c acc => constr_def_value_wop match_ind c *m acc) unit) match_constrs_sig;
    match_param_args: ArgsOf match_param_tele;
    match_return_predicate: return_predicate_type match_param_args match_ind match_sort;
  }.
```
c.f. https://github.com/coq/coq/issues/11248#issuecomment-562650626

In this case, the issue is that the minimizer regex was overly sensitive to how many universes had been declared due to spacing issues.

--- a/find-bug.py
+++ b/find-bug.py
@@ -904,8 +904,13 @@ Tactic Notation "admit" := abstract case proof_admitted.
 End AdmitTactic.
 """
     tac_code_re = r"""\s*Module Export AdmitTactic\.
-.*Tactic Notation "admit" := abstract case proof_admitted\.
-\s*End AdmitTactic\.\n*"""
+?(?:Module Import LocalFalse\.
+?(?:Inductive False := \.)?
+?End LocalFalse\.)?
+?(?:Axiom proof_admitted : False\.)?
+?(?:Import Coq.Init.Notations\.)?
+?(?:Tactic Notation "admit" := abstract case proof_admitted\.)?
+?End AdmitTactic\.\n*"""
     return '%s%s' % (tac_code, re.sub(tac_code_re, '\n', contents.replace('Require Coq.Init.Notations.\n', ''), flags=re.DOTALL|re.MULTILINE))
 
 

